.github/workflows/buildAndTest.yml

```diff
@@ -338,3 +345,102 @@ jobs:
+
+      - name: Ccache for C++ compilation
+        # https://github.com/hendrikmuhs/ccache-action/releases/tag/v1.2.9
+        uses: hendrikmuhs/ccache-action@ca3acd2731eef11f1572ccb126356c2f9298d35e
+        with:
+          key: ${{ runner.os }}-${{ matrix.ubuntu_version }}-${{ steps.get-llvm-commit-hash.outputs.hash }}
+          max-size: 1G
+
+      - name: Build and install LLVM
+        run: LLVM_ENABLE_RTTI=ON utils/build-llvm.sh
+
+      - name: Install our python reqs
+        run: pip install -r python/requirements.txt
+
+      - name: Build and generate coverage (Release)
+        run: |
+          mkdir build_release
+          cd build_release
+          cmake .. \
+            -GNinja \
+            -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ \
+            -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+            -DCMAKE_EXE_LINKER_FLAGS_INIT="-fuse-ld=lld" -DCMAKE_MODULE_LINKER_FLAGS_INIT="-fuse-ld=lld" -DCMAKE_SHARED_LINKER_FLAGS_INIT="-fuse-ld=lld" \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DCMAKE_PLATFORM_NO_VERSIONED_SONAME=ON \
+            -DCMAKE_VISIBILITY_INLINES_HIDDEN=ON \
+            -DCMAKE_C_VISIBILITY_PRESET=hidden \
+            -DCMAKE_CXX_VISIBILITY_PRESET=hidden \
+            -DAIE_COMPILER=NONE \
+            -DAIE_LINKER=NONE \
+            -DHOST_COMPILER=NONE \
+            -DBUILD_INSTRUMENTED_COVERAGE=ON \
+            -DLLVM_ENABLE_ASSERTIONS=OFF \
+            -DLLVM_ENABLE_RTTI=ON \
+            -DCMAKE_MODULE_PATH=`pwd`/../cmake/modulesXilinx \
+            -DMLIR_DIR=/home/runner/work/mlir-aie/mlir-aie/llvm/install/lib/cmake/mlir \
+            -DLLVM_DIR=/home/runner/work/mlir-aie/mlir-aie/llvm/install/lib/cmake/llvm \
+            -DLLVM_USE_LINKER=lld \
+            -DLLVM_EXTERNAL_LIT=`pwd`/../llvm/build/bin/llvm-lit
+          
+          ninja && ninja generate-aie-coverage-report
+
+      - name: Format coverage report
+        id: format-report
+        run: |
+          sed -i.bak 's/<!doctype html>/<!--<!doctype codecov html>-->/g' /home/runner/work/mlir-aie/mlir-aie/build_release/report/index.html
+          sed -i.bak 's/<pre>//g' /home/runner/work/mlir-aie/mlir-aie/build_release/report/index.html
+          sed -i.bak 's/<\/pre>//g' /home/runner/work/mlir-aie/mlir-aie/build_release/report/index.html
+          sed -i.bak 's/([0-9]*\/[0-9]*)//g' /home/runner/work/mlir-aie/mlir-aie/build_release/report/index.html
+          sed -i.bak "s/href=/href=''/g" /home/runner/work/mlir-aie/mlir-aie/build_release/report/index.html
+
+
+      - name: Update PR with coverage results
+        uses: edumserrano/find-create-or-update-comment@v2
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body-includes: '<!--<!doctype codecov html>-->'
+          comment-author: 'github-actions[bot]'
+          body-path: /home/runner/work/mlir-aie/mlir-aie/build_release/report/index.html
+          edit-mode: replace
```

Comment:
Is this better than having a build artifact?

Reply:
In particular, would be interested in your thought on how to make this more actionable?  For instance, is there a way to link to annotated source code with llvm-cov, so we can figure out which lines aren't covered?

Reply:
There is a [build artifact as well](https://gist.github.com/makslevental/c9462d50816727f9f3e9a5f5e0dedbd7) that could be `cat`ed 
<img width="205" alt="image" src="https://github.com/Xilinx/mlir-aie/assets/5657668/7dbc2ba8-c5d9-4c8b-9917-6bbafe12f158">
but it would take more fiddling with `cut` then I'm willing to do to 🤷. 

The value of appending coverage for all source files is marginal I admit so I'm fiddling right now to see if I can narrow it to just files touched by the PR.

Reply:
> For instance, is there a way to link to annotated source code with llvm-cov, so we can figure out which lines aren't covered?

If you genereate the report locally then yes you can click each source file and get line-by-line stats. 

>  In particular, would be interested in your thought on how to make this more actionable?

Honor system: if you're adding code and you didn't add tests, this will at least be a loud reminder.